### PR TITLE
Date Class Implementation (Partial)

### PR DESCRIPTION
**Pull request by BryanMorfe**
Started implementation of Date class (unfinished) -- not compiled to test.

### CODE REVIEW

File: source/Date.h [hunk 1]
```diff
@@ -2,25 +2,37 @@
 #ifndef DATE_H
 #define DATE_H
 
-#include <ctime>
+#include <stdint.h>
```
Comment (kefsaj) [inline source/Date.h:5]: Why are we including this? Is it really necessary?

File: source/Date.h [hunk 2]
```diff
@@ -2,25 +2,37 @@
...
 class Date
 {
-    
 private:
-    int DAY;
-    int MONTH;
-    int YEAR;
+    uint8_t DAY;
+    uint8_t MONTH;
+    uint8_t YEAR;
```
Comment (kefsaj) [inline source/Date.h]: Year must be at least 16 bits.

File: source/Date.h [hunk 3]
```diff
@@ -2,25 +2,37 @@
...
     
 public:
-    Date()= default;
-    Date(const int& D, const int& M, const int& Y);
-    now();
+    Date() = default;
+    Date(uint8_t D, uint8_t M, uint8_t Y);
```
Comment (kefsaj) [inline source/Date.h]: Again, year has to have at least 16 bits.

File: source/Date.h [hunk 4]
```diff
@@ -2,25 +2,37 @@
...
+    
+    /* Getters */
+    uint8_t getDAY() const;
+    uint8_t getMONTH() const;
+    uint8_t getYEAR() const;
```
Comment (kefsaj) [inline source/Date.h:21]: Ditto.

File: source/Date.h [hunk 5]
```diff
@@ -2,25 +2,37 @@
...
     
-    /* Getters*/
-    Date getDate() const;
+    /* Setters */
+    void setDAY(uint8_t D);
+    void setMONTH(uint8_t M);
+    void setYEAR(uint8_t Y);
```
Comment (kefsaj) [inline source/Date.h]: These are covered by the setDate function, so they must be removed.

File: source/Date.h [hunk 6]
```diff
@@ -2,25 +2,37 @@
...
+    void setDate(uint8_t D, uint8_t M, uint8_t Y);
```
Comment (kefsaj) [inline source/Date.h]: Year, 16 bits.

File: source/Date.cpp [hunk 7]
```diff
@@ -1,23 +1,72 @@
...
+    if(Y < 1970 || Y > 2039 || M > 12 || D > 31) {
+        throw std::logic_error("Invalid date range");
+    }
+    
+    DAY = D;
+    MONTH = M;
+    YEAR = Y;
 }
 
+
```
Comment (kefsaj) [inline source/Date.cpp:18]: Extra space not needed. It must be removed.

File: source/Date.cpp [hunk 8]
```diff
@@ -1,23 +1,72 @@
...
-    
+uint8_t Date::getDAY() const {
+    return DAY;
+}
+
+uint8_t Date::getMONTH() const {
+    return MONTH;
+}
+
+uint8_t Date::getYEAR() const {
```
Comment (kefsaj) [inline source/Date.cpp:28]: Remember to change to 16 bits.

File: source/Date.cpp [hunk 9]
```diff
@@ -1,23 +1,72 @@
...
-{}
-now(){
-    auto timenow =
-         chrono::system_clock::to_time_t(chrono::system_clock::now());
-     
-       cout << ctime(&timenow) << endl;
+#include <ctime>
+#include <stdexcept>
+
+Date::Date(uint8_t D, uint8_t M, uint8_t Y) {
```
Comment (kefsaj) [inline source/Date.cpp]: Remember to change to 16 bits.

File: source/Date.cpp [hunk 10]
```diff
@@ -1,23 +1,72 @@
...
+    }
+    
+    DAY = D;
+    MONTH = M;
+    YEAR = Y;
+    
+}
+
+/* Operators */
+Date& Date::operator+(const Date& other) {
```
Comment (kefsaj) [inline source/Date.cpp:46]: Not implemented... Should be noted in a comment

File: source/Date.cpp [hunk 11]
```diff
@@ -1,23 +1,72 @@
...
+    
+    if(DAY + other.DAY > 12) {
+        MONTH = 1 + (MONTH + 1) % 13;
+    }
+    
+    DAY = 1 + (DAY + other.DAY) % 32;
+    
+    if(MONTH )
```
Comment (kefsaj) [inline source/Date.cpp]: This line should be commented also

File: source/Date.cpp [hunk 12]
```diff
@@ -1,23 +1,72 @@
...
+    
+    YEAR += other.YEAR;
+    
+    return *this;
+}
+
+Date& Date::operator-(const Date& other) {
+
```
Comment (kefsaj) [inline source/Date.cpp:64]: If not implemented, should be noted in a comment.

**Review by kefsaj — CHANGES_REQUESTED** [review]
Change as needed.

File: source/Date.cpp (hunk 12 above)
Comment (BryanMorfe) [reply source/Date.cpp:64]: Done.

File: source/Date.cpp (hunk 11 above)
Comment (BryanMorfe) [reply source/Date.cpp]: Done.

File: source/Date.cpp (hunk 10 above)
Comment (BryanMorfe) [reply source/Date.cpp:46]: Done.

File: source/Date.h (hunk 1 above)
Comment (BryanMorfe) [reply source/Date.h:5]: Yes, it is to use uint8_t and uint16_t

File: source/Date.h (hunk 5 above)
Comment (BryanMorfe) [reply source/Date.h]: Done.

File: source/Date.cpp (hunk 8 above)
Comment (BryanMorfe) [reply source/Date.cpp:28]: Done.

File: source/Date.cpp (hunk 9 above)
Comment (BryanMorfe) [reply source/Date.cpp]: Done.

**Review by kefsaj — APPROVED** [review]

**Pull request merged**